sync_cleared_for_campus.R

```diff
@@ -0,0 +1,179 @@
+    synced_dataset <- dataset %>%
+        filter( (!!ta != !!tb) | xor(is.na(!!ta), is.na(!!tb)) ) %>%
+        mutate(!!timestamp_field := case_when(
+                    ymd_hms(!!ta) > ymd_hms(!!tb) ~ !!ta,
+                    ymd_hms(!!tb) > ymd_hms(!!ta) ~ !!tb,
+                    is.na(!!ta) ~ !!tb,
+                    is.na(!!tb) ~ !!ta,
+                    TRUE ~ !!ta # default to Result in event of identical timestamps or unknown case
+                    )
+              ) %>%
+        mutate(!!field_to_sync := case_when(
+                    ymd_hms(!!ta) > ymd_hms(!!tb) ~ !!fa,
+                    ymd_hms(!!tb) > ymd_hms(!!ta) ~ !!fb,
+                    is.na(!!ta) ~ !!fb,
+                    is.na(!!tb) ~ !!fa,
+                    TRUE ~ !!fa # default to Result in event of identical timestamps or unknown case
+                    )
+              )
+
+    # hack variable name strings for second pass
+    timestamp_field.x = paste0(timestamp_field, '.x')
+    timestamp_field.y = paste0(timestamp_field, '.y')
+    tx = sym(timestamp_field.x)
+    ty = sym(timestamp_field.y)
+
+    field_to_sync.x = paste0(field_to_sync, '.x')
+    field_to_sync.y = paste0(field_to_sync, '.y')
+    fx = sym(field_to_sync.x)
+    fy = sym(field_to_sync.y)
+
+    # create dataframe of corrections for projects a and b
+    # suffixes are '.x' for project dataset and '.y' for synced dataset
+
+    writes_a <- dataset_a %>%
+        left_join(synced_dataset, by = join_field_in_a) %>%
+        filter( (!!tx < !!ty) | (is.na(!!tx) & !is.na(!!ty)) ) %>%
+        mutate(!!timestamp_field := !!ty) %>%
+        mutate(!!field_to_sync := !!fy)
+
+    writes_b <- dataset_b %>%
+        left_join(synced_dataset, by = setNames(nm=join_field_in_b, join_field_in_a)) %>%
+        filter( (!!tx < !!ty) | (is.na(!!tx) & !is.na(!!ty)) ) %>%
+        mutate(!!timestamp_field := !!ty) %>%
+        mutate(!!field_to_sync := !!fy)
+
+    return(list(writes_a, writes_b))
+}
+
+W <- sync_fields_by_timestamp(dataset_a = result_records, join_field_in_a = 'record_id',
+        dataset_b = case_records, join_field_in_b = 'case_emp_ufid',
+        timestamp_field = 'cleared_for_campus_timestamp', field_to_sync = 'cleared_for_campus')
+
+# not using zeallot's %<-% as this will be called regularly enough to consider the lib load overhead
+result_rewrites <- W[[1]] %>%
+    select(record_id, cleared_for_campus, cleared_for_campus_timestamp)
+case_rewrites <- W[[2]] %>%
+    select(record_id, redcap_event_name = redcap_event_name.x, cleared_for_campus, cleared_for_campus_timestamp) %>%
+    unique() %>%
+    group_by(record_id) %>% # in event of duplicate record_id, keep only row with latest timestamp
+    top_n(1, cleared_for_campus_timestamp) %>%
+    ungroup()
+
+W <- sync_fields_by_timestamp(dataset_a = result_records, join_field_in_a = 'record_id',
+        dataset_b = contact_records, join_field_in_b = 'case_emp_ufid',
```

Comment:
```suggestion
        dataset_b = contact_records, join_field_in_b = 'contact_ll_ufid',
```
I think this is correct, we're looking for the _contact_'s information, right?